### PR TITLE
feat(file-uploader): Prop to turn off or on images in Tracker

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
This new prop allows customers to turn on or off images to display in the tracker component

#### Issue #, if available
[Asana](https://app.asana.com/0/1203303459000520/1203373828802426)

#### Description of how you validated changes
Add test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx [hunk 1]
```diff
@@ -357,23 +358,23 @@ export function FileUploader({
...
-            url={URL.createObjectURL(status.file)}
+            fileState={status?.fileState}
+            hasImage={status.file?.type.startsWith('image/') && showImages}
             key={index}
-            onChange={onNameChange(index)}
+            name={status.name}
             onCancel={onFileCancel(index)}
+            onCancelEdit={onCancelEdit(index)}
+            onChange={onNameChange(index)}
+            onDelete={onDelete}
```
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/FileUploader.tsx:370]: I resorted these alphabetically

File: packages/react/src/components/Storage/FileUploader/Tracker/index.tsx [hunk 2]
```diff
@@ -20,20 +20,20 @@ import {
...
 export function Tracker({
+  errorMessage,
   file,
   fileState,
   hasImage,
-  url,
+  name,
+  onCancel,
+  onCancelEdit,
   onPause,
```
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/Tracker/index.tsx:30]: Resorted Alphabetically

File: packages/react/src/components/Storage/FileUploader/types.ts [hunk 3]
```diff
@@ -62,22 +63,22 @@ export interface PreviewerProps extends DragActionHandlers {
...
   fileState: FileState;
   hasImage: boolean;
-  url: string;
-  resumable?: boolean;
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  name: string;
   onCancel: () => void;
+  onCancelEdit?: () => void;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onDelete?: () => void;
```
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/types.ts:74]: Resorted alphabetically too , that's the only change here

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx [hunk 4]
```diff
@@ -357,23 +358,23 @@ export function FileUploader({
       >
         {fileStatuses?.map((status, index) => (
           <Tracker
-            percentage={status.percentage}
+            errorMessage={status?.fileErrors}
             file={status.file}
-            hasImage={status.file?.type.startsWith('image/')}
-            url={URL.createObjectURL(status.file)}
+            fileState={status?.fileState}
+            hasImage={status.file?.type.startsWith('image/') && showImages}
```
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/FileUploader.tsx:364]: This the change I added. So now it has to have an image and showImages has to be true.

Comment (dbanksdesign) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:364]: I think we need to change this logic a bit. I was thinking this prop would show or hide both the image and file icon depending if it is true or false. If `showImages` is true, there will always be either an image or icon. If `showImages` is false, no image or icon will appear. Does that make sense?

**Review by dbanksdesign — COMMENTED** [review]
Looks good, just one clarification

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx (hunk 4 above)
Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:364]: OK! @dbanksdesign I can make it so it doesn't show the icon either!

Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx]: Fixed in [cf8a443](https://github.com/aws-amplify/amplify-ui/pull/2988/commits/cf8a443a1d574255cda35fa8fdb5f8a3b4585418)

File: packages/react/src/components/Storage/FileUploader/Tracker/index.tsx [hunk 5]
```diff
@@ -131,7 +132,13 @@ export function Tracker({
...
   return (
     <View className={ComponentClassNames.FileUploaderFile}>
-      <View className={ComponentClassNames.FileUploaderFileImage}>{icon}</View>
+      {showImage ? (
+        <View className={ComponentClassNames.FileUploaderFileImage}>
+          {icon}
+        </View>
+      ) : (
+        ''
+      )}
```
Comment (dbanksdesign) [inline packages/react/src/components/Storage/FileUploader/Tracker/index.tsx:141]:
```suggestion
      ) : null}
```

**Review by dbanksdesign — APPROVED** [review]
One non-blocking nit. LGTM!

File: packages/react/src/components/Storage/FileUploader/Tracker/index.tsx (hunk 5 above)
Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/Tracker/index.tsx]: OK! I'll fix it really quickly

**Review by 0618 — APPROVED** [review]

**Pull request merged**